Cross node connectivity fails with IPsec +  tunneling + endpoint routes

Tried to enable the following configuration in the https://github.com/cilium/cilium/pull/22825:
```
name: '3'
kernel: '5.10-main'
kube-proxy: 'iptables'
kpr: 'disabled'
tunnel: 'vxlan'
encryption: 'ipsec'
endpoint-routes: 'true'
```

Unfortunately, the connectivity test failed with that configuration (https://github.com/cilium/cilium/actions/runs/3909668341/jobs/6686245297):
```
 Deployment        cilium-operator    Desired: 1, Ready: 1/1, Available: 1/1
DaemonSet         cilium             Desired: 3, Ready: 3/3, Available: 3/3
Containers:       cilium             Running: 3
                  cilium-operator    Running: 1
Cluster Pods:     3/3 managed by Cilium
Image versions    cilium-operator    quay.io/cilium/operator-generic-ci:647a726040a30492f06e0bc6db612ac87d4c36e1: 1
                  cilium             quay.io/cilium/cilium-ci:647a726040a30492f06e0bc6db612ac87d4c36e1: 3
✨ [kind-kind] Creating namespace cilium-test for connectivity check...
✨ [kind-kind] Deploying echo-same-node service...
✨ [kind-kind] Deploying DNS test server configmap...
✨ [kind-kind] Deploying same-node deployment...
✨ [kind-kind] Deploying client deployment...
✨ [kind-kind] Deploying client2 deployment...
✨ [kind-kind] Deploying echo-other-node service...
✨ [kind-kind] Deploying other-node deployment...
✨ [kind-kind] Deploying host-netns daemonset...
⌛ [kind-kind] Waiting for deployments [client client2 echo-same-node] to become ready...
⌛ [kind-kind] Waiting for deployments [echo-other-node] to become ready...
⌛ [kind-kind] Waiting for CiliumEndpoint for pod cilium-test/client-755fb678bd-llgxm to appear...
⌛ [kind-kind] Waiting for CiliumEndpoint for pod cilium-test/client2-5b97d7bc66-4bnpw to appear...
⌛ [kind-kind] Waiting for pod cilium-test/client-755fb678bd-llgxm to reach DNS server on cilium-test/echo-same-node-64774c64d5-p5xg5 pod...
⌛ [kind-kind] Waiting for pod cilium-test/client2-5b97d7bc66-4bnpw to reach DNS server on cilium-test/echo-same-node-64774c64d5-p5xg5 pod...
⌛ [kind-kind] Waiting for pod cilium-test/client-755fb678bd-llgxm to reach DNS server on cilium-test/echo-other-node-67b74b6685-6l5dc pod...
connectivity test failed: timeout reached waiting lookup for localhost from pod cilium-test/client-755fb678bd-llgxm to server on pod cilium-test/echo-other-node-67b74b6685-6l5dc to succeed (last error: context deadline exceeded)
Error: Process completed with exit code 1.
```

The sysdump: [cilium-sysdump-5.10-main-final.zip](https://github.com/cilium/cilium/files/10413115/cilium-sysdump-5.10-main-final.zip)

--- a/pkg/datapath/linux/ipsec/ipsec_linux.go
+++ b/pkg/datapath/linux/ipsec/ipsec_linux.go
@@ -196,11 +196,11 @@ func _ipSecReplacePolicyInFwd(src, dst *net.IPNet, tmplSrc, tmplDst net.IP, prox
 	policy := ipSecNewPolicy()
 	policy.Dir = dir
 	policy.Dst = dst
-	policy.Mark = &netlink.XfrmMark{
-		Mask: linux_defaults.IPsecMarkMaskIn,
-	}
 	if dir == netlink.XFRM_DIR_IN {
 		policy.Src = src
+		policy.Mark = &netlink.XfrmMark{
+			Mask: linux_defaults.IPsecMarkMaskIn,
+		}
 		if proxyMark {
 			// We require a policy to match on packets going to the proxy which are
 			// therefore carrying the proxy mark. We however don't need a policy

--- a/test/k8s/datapath_configuration.go
+++ b/test/k8s/datapath_configuration.go
@@ -216,6 +216,23 @@ var _ = Describe("K8sDatapathConfig", func() {
 			}
 		})
 
+		SkipItIf(func() bool {
+			// IPsec + encapsulation requires Linux 4.19.
+			// We also can't disable KPR on GKE at the moment (cf. #16597).
+			return helpers.RunsWithoutKubeProxy() || helpers.DoesNotRunOn419OrLaterKernel() || helpers.RunsOnGKE() || helpers.RunsOnAKS()
+		}, "Check connectivity with transparent encryption, VXLAN, and endpoint routes", func() {
+			deploymentManager.Deploy(helpers.CiliumNamespace, IPSecSecret)
+			options := map[string]string{
+				"kubeProxyReplacement":   "disabled",
+				"encryption.enabled":     "true",
+				"endpointRoutes.enabled": "true",
+			}
+			enableVXLANTunneling(options)
+			deploymentManager.DeployCilium(options, DeployCiliumOptionsAndDNS)
+			validateBPFTunnelMap()
+			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test with IPsec between nodes failed")
+		}, 600)
+
 		SkipItIf(helpers.SkipQuarantined, "Check iptables masquerading with random-fully", func() {
 			options := map[string]string{
 				"bpf.masquerade":       "false",

